src/pinning.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2019, QuantStack and Mamba Contributors
+//
+// Distributed under the terms of the BSD 3-Clause License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+
+#include "mamba/pinning.hpp"
+
+#include <fstream>
+
+
+namespace mamba
+{
+    std::string python_pin(const PrefixData& prefix_data, const std::vector<std::string>& specs)
+    {
+        std::string pin = "";
+        std::string py_version;
+        MatchSpec ms;
+
+        try
```

Comment:
hmm, I am more a fan of (from stackoverflow):

```
MyMapType::const_iterator iter = mymap.find(key);

if(iter != mymap.end())
{
    // iter is item pair in the map. The value will be accessible as `iter->second`.
}
```

Reply:
Thanks again for the suggestion!